cmd/checkmarxExecuteScan.go

```diff
@@ -347,24 +350,26 @@ func enforceThresholds(config checkmarxExecuteScanOptions, results map[string]in
 	return insecure
 }
 
-func createAndConfigureNewProject(sys checkmarx.System, projectName, teamID, presetValue, engineConfiguration string) checkmarx.Project {
-	ok, projectCreateResult := sys.CreateProject(projectName, teamID)
-	if ok {
-		if len(presetValue) > 0 {
-			ok, preset := loadPreset(sys, presetValue)
-			if ok {
-				configurationUpdated := sys.UpdateProjectConfiguration(projectCreateResult.ID, preset.ID, engineConfiguration)
-				if configurationUpdated {
-					log.Entry().Debugf("Configuration of project %v updated", projectName)
-				} else {
-					log.Entry().Fatalf("Updating configuration of project %v failed", projectName)
-				}
+func updateProjectPreset(sys checkmarx.System, projectID int, presetValue, engineConfiguration string) {
+	if len(presetValue) > 0 {
+		ok, preset := loadPreset(sys, presetValue)
+		if ok {
+			configurationUpdated := sys.UpdateProjectConfiguration(projectID, preset.ID, engineConfiguration)
+			if configurationUpdated {
+				log.Entry().Debugf("Configuration of project %v updated", projectID)
 			} else {
-				log.Entry().Fatalf("Preset %v not found, creation of project %v failed", presetValue, projectName)
+				log.Entry().Fatalf("Updating configuration of project %v failed", projectID)
 			}
 		} else {
-			log.Entry().Fatalf("Preset not specified, creation of project %v failed", projectName)
+			log.Entry().Fatalf("Preset %v not found, creation of project %v failed", presetValue, projectID)
 		}
+	} else {
+		log.Entry().Fatalf("Preset not specified, creation of project %v failed", projectID)
+	}
```

Comment:
The function `updateProjectPreset()` is called unconditionally from `runScan()`. If no `preset` is given, then we would exit here and fail the pipeline. Also, the log-message is misleading, if the project would only have to be updated.